MaxText/layers/attentions.py

```diff
@@ -335,6 +370,10 @@ def generate_attention_mask(
       chunk_mask = _generate_chunk_attention_mask(mask_shape=(q_seq_len, kv_seq_len), chunk_size=self.chunk_attn_window_size)
       output_mask = chunk_mask * output_mask
 
+    if bidirectional_mask is not None:
+      image_mask = _make_bidirectional_block_mask(bidirectional_mask)
+      output_mask = output_mask | image_mask[:, None, None, ...]
```

Comment:
Does output_mask need to be extended? 
e.g. if sequence length is 6, output_mask is 6x6 ? do you need to extend it to count for image token position?
It would be great to have some tests of combining text mask + image mask.

Reply:
output_mask is already broadcasted to (1,1,1,seq_len, seq_len). So only extend image_mask to (batch, 1, 1, seq_len, seq_len). I added a unit test: test_combine_with_causal_mask